code/modules/shuttles/shuttle_supply_multi.dm

```diff
@@ -0,0 +1,825 @@
+						shaft_projectors[area_of_T.fake_zlevel] |= P
+
+	if(supply_controller)
+		for(var/i=1; i <= length(GLOB.supply_elevator_turfs);i++)
+			var/turf/T = GLOB.supply_elevator_turfs[i]
+			var/area/T_area = get_area(T)
+			var/Ts_fake_zlevel = T_area.fake_zlevel
+
+			elevator_effects["[Ts_fake_zlevel]SW"]  = new /obj/effect/elevator/supply/multi(locate(T.x-2,T.y-2,T.z))
+			elevator_effects["[Ts_fake_zlevel]SW"].vis_contents += elevator_animationz[Ts_fake_zlevel]
+
+			elevator_effects["[Ts_fake_zlevel]SE"] = new /obj/effect/elevator/supply/multi(locate(T.x+2,T.y-2,T.z))
+			elevator_effects["[Ts_fake_zlevel]SE"].pixel_x = -128
+			elevator_effects["[Ts_fake_zlevel]SE"].vis_contents += elevator_animationz[Ts_fake_zlevel]
+
+			elevator_effects["[Ts_fake_zlevel]NW"] = new /obj/effect/elevator/supply/multi(locate(T.x-2,T.y+2,T.z))
+			elevator_effects["[Ts_fake_zlevel]NW"].pixel_y = -128
+			elevator_effects["[Ts_fake_zlevel]NW"].vis_contents += elevator_animationz[Ts_fake_zlevel]
+
+			elevator_effects["[Ts_fake_zlevel]NE"] = new /obj/effect/elevator/supply/multi(locate(T.x+2,T.y+2,T.z))
+			elevator_effects["[Ts_fake_zlevel]NE"].pixel_x = -128
+			elevator_effects["[Ts_fake_zlevel]NE"].pixel_y = -128
+			elevator_effects["[Ts_fake_zlevel]NE"].vis_contents += elevator_animationz[Ts_fake_zlevel]
+
+	var/list/all_directions = list(NORTH, SOUTH, EAST, WEST, NORTHEAST, NORTHWEST, SOUTHEAST, SOUTHWEST)
+
+	for(var/turf/T in shaft_transit_area)
+		for(var/i in all_directions)
+			if(istype(get_step(T, i), /turf/closed/wall/supply_shaft))
+				transit_location_walls |= get_step(T, i)
+
+/datum/shuttle/ferry/supply/multi/add_structures()  //places the shuttle controller on the lift
+	for(var/turf/T in area_offsite)
+		var/check1 = (get_step(T, NORTH).density ? TRUE : FALSE)
+		var/check2 = (get_step(T, EAST).density ? TRUE : FALSE)
+		if(check1 && check2)
+			in_elevator_controller = new/obj/structure/elevator_control_mount(T)
+			break
+
+/datum/shuttle/ferry/supply/multi/pick_loc()
+	RETURN_TYPE(/turf)
+	if(!supply_controller.multi_fakezlevel_elevator)
+		return GLOB.supply_elevator_turfs[target_zlevel]
+
+/datum/shuttle/ferry/supply/multi/get_location_area(location_id)
+	if(moving_status != SHUTTLE_INTRANSIT)
+		if(!fake_zlevel)
+			return area_offsite
+		else
+			return get_area(GLOB.supply_elevator_turfs[fake_zlevel])
+	else
+		return shaft_transit_area
+
+/datum/shuttle/ferry/supply/multi/proc/get_movement_direction()
+	//0 means DOWN; 1 means UP
+	if(moving_status == SHUTTLE_IDLE)
+		return null
+
+	if(!target_zlevel)
+		return 0
+
+	else if(!fake_zlevel)
+		return 1
+
+	else if(target_zlevel < fake_zlevel)
+		return 1
+	else
+		return 0
+
+/datum/shuttle/ferry/supply/multi/at_station()
+	return fake_zlevel
+
+/datum/shuttle/ferry/supply/multi/short_jump(area/origin, area/destination)
+	if(moving_status != SHUTTLE_IDLE)
+		return
+
+	if(isnull(location) || target_zlevel == fake_zlevel)
+		return
+
+	recharging = 1
```

Comment:
is this a bool? use TRUE

Reply:
it aint, its rounded in long_jump()??? shuttle code jank

Reply:
what the hell